sqlalchemy_utils/functions/database.py

```diff
@@ -422,21 +422,17 @@ def is_auto_assigned_date_column(column):
 
 
 def _set_url_database(url: sa.engine.url.URL, database):
-    if hasattr(sa.engine, 'URL'):
-        ret = sa.engine.URL.create(
-            drivername=url.drivername,
-            username=url.username,
-            password=url.password,
-            host=url.host,
-            port=url.port,
-            database=database,
-            query=url.query
-        )
-    else:  # SQLAlchemy <1.4
-        url.database = database
-        ret = url
-    assert ret.database == database, ret
-    return ret
+    """Set the database of an engine URL.
+    
+    :param url: A SQLAlchemy engine URL.
+    :param database: New database to set.
+    
+    """
+    if hasattr(url, 'set'):  # SQLAlchemy >1.4
+        return url.set(database=database)
```

Comment:
This won't reset the database field if `database` is None, we already tried that.

Reply:
Ah. Is that why it was necessary to recreate it?

If that's the case, couldn't we utilize the same check, but recreate it instead? Going to try that real quick to see if the extra failing tests pass.

Reply:
Okay. That *seems* to have solved the problem. At least, the extra failed tests are passing now anyways.